src/inc/msquic.h

```diff
@@ -1605,6 +1605,62 @@ QUIC_STATUS
     _In_opt_ void* ClientSendContext
     );
 
+//
+// Connection Pool API
+//
+
+//
+// Closes the Connection Pool API table and cleans up the API reference.
+//
+typedef
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+(QUIC_API * QUIC_CONN_POOL_API_CLOSE)(
+    _In_ void* ConnPoolApi
+    );
+
+//
+// Creates a simple connection pool with NumberOfConnections connections
+// all with the same Context and Handler, and puts them in the
+// caller-supplied array.
+//
+// One of ServerName or ServerAddress *MUST* be supplied, and for the pool to
+// work correctly, the connections *MUST* be started with the same ServerName
+// and ServerPort as supplied in this call.
+//
+typedef
+_IRQL_requires_max_(PASSIVE_LEVEL)
+_Check_return_
+QUIC_STATUS
+(QUIC_API * QUIC_SIMPLE_CONN_POOL_CREATE_FN)(
+    _In_ HQUIC* Registration,
+    _In_ QUIC_CONNECTION_CALLBACK_HANDLER Handler,
+    _In_opt_ void* Context,
+    _In_opt_ const char* ServerName,
+    _In_opt_ const QUIC_ADDR* ServerAddress,
+    _In_ uint16_t ServerPort,
+    _In_ uint32_t NumberOfConnections,
+    _Out_writes_bytes_(NumberOfConnections * sizeof(HQUIC))
+        HQUIC** ConnectionPool
+    );
```

Comment:
I think the extra API table is unnecessary. Let's just add this function directly to the MsQuic table and eliminate all the rest.

Reply:
But with the second level API table, we can extend the connection pool API to more than just an array of connections. I don't know how future-looking we should be with this design

Reply:
Let's keep it as simple as possible. As long as it's wrapped by preview flag we can change later if necessary.